internal/pkg/cli/pipeline_deploy.go

```diff
@@ -180,17 +193,30 @@ func (o *deployPipelineOpts) Execute() error {
 }
 
 func (o *deployPipelineOpts) validatePipelineName() error {
```

Comment:
For `pipeline deploy` we look for the name in pipeline manifests because they may have only run `pipeline init` and the pipeline is not yet deployed.
For `pipeline delete`, we can only delete pipelines that have been deployed, so we look for deployed pipelines instead of local ones.